.github/workflows/built-image.yml

```
@@ -0,0 +1,133 @@
+name: Build and Push Docker Image
+
+on:
+  workflow_call:
+    inputs:
+      working_directory:
+        type: string
+        description: What directory should the image be built from
+        required: true
+      image_name:
+        type: string
+        description: Name of Docker image
+        required: true
+      image_tag:
+        type: string
+        description: Tag for Docker image
+        required: true
+      push_image:
+        type: boolean
+        description: Should the image be pushed to the Github Container Registry
+        required: false
+        default: false
+    outputs:
+      image_name:
+        description: Fully qualified image name
+        value: ${{ jobs.build-image.outputs.image_name}}
+      image_ref:
+        description: Docker image reference
+        value: ${{ jobs.build-image.outputs.image_ref }}
+
+  workflow_dispatch:
+    inputs:
+      working_directory:
+        type: string
+        description: What directory should the image be built from
+        required: true
+      image_name:
+        type: string
+        description: Name of Docker image
+        required: true
+      image_tag:
+        type: string
+        description: Tag for Docker image
+        required: true
+      push_image:
+        type: boolean
+        description: Should the image be pushed to the Github Container Registry
+        required: false
+        default: false
+
+jobs:
+  build-image:
+    runs-on: ubuntu-22.04
+    name: Build and push image
+    timeout-minutes: 30
+
+    outputs:
+      image_name: ${{ steps.env_var.outputs.image_name }}
+      image_ref: ${{ steps.env_var.outputs.image_ref }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1.5.1
```

The checkout one I do recommend the bump, this one I'm not sure.

```suggestion
        uses: docker/setup-buildx-action@v2
```